File: leetcode/AIcoding/2.Gemini-2.0/python3_solutions/3000-maximum-area-of-longest-diagonal-rectangle.py
import math

class Solution:
    def areaOfMaxDiagonal(self, dimensions: list[list[int]]) -> int:
        max_diagonal = 0
        max_area = 0
        for l, w in dimensions:
            diagonal = math.sqrt(l * l + w * w)
            area = l * w
            if diagonal > max_diagonal:
                max_diagonal = diagonal
                max_area = area
            elif diagonal == max_diagonal:
                max_area = max(max_area, area)
        return max_area

File: leetcode/AIcoding/3.DeepSeek-v3/javascript_solutions/3000-maximum-area-of-longest-diagonal-rectangle.js
var areaOfMaxDiagonal = function(dimensions) {
    let maxDiagonal = 0;
    let maxArea = 0;

    for (const [l, w] of dimensions) {
        const diagonal = l * l + w * w;
        if (diagonal > maxDiagonal) {
            maxDiagonal = diagonal;
            maxArea = l * w;
        } else if (diagonal === maxDiagonal) {
            maxArea = Math.max(maxArea, l * w);
        }
    }

    return maxArea;
};